src/peergos/shared/user/fs/TagIndex.java

```diff
@@ -0,0 +1,75 @@
+package peergos.shared.user.fs;
+
+import peergos.shared.cbor.CborObject;
+import peergos.shared.cbor.Cborable;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class TagIndex  implements Cborable {
+
+    private final Map<String, Set<String>> entries;
+
+    public TagIndex(Map<String, Set<String>> entries) {
+        this.entries = entries;
+    }
+
+    public void addEntry(String tagName, Set<String> fileList) {
+        if(entries.containsKey(tagName)) {
+            this.entries.get(tagName).addAll(fileList);
+        } else {
+            this.entries.put(tagName, fileList);
+        }
+    }
+
+    public void removeEntry(String tagName, Set<String> fileList) {
+        if(entries.containsKey(tagName)) {
+            this.entries.get(tagName).removeAll(fileList);
+            if(this.entries.get(tagName).isEmpty()) {
+                this.entries.remove(tagName);
+            }
+        }
+    }
+
+    public TagIndex addEntries(Map<String, Set<String>> newEntries) {
+        newEntries.forEach(this::addEntry);
+        return this;
+    }
+
+    public TagIndex removeEntries(Map<String, Set<String>> toRemoveEntries) {
+        toRemoveEntries.forEach(this::removeEntry);
+        return this;
+    }
+
+    public Map<String, Set<String>> getEntries() {
+        return entries;
+    }
+
+    @Override
+    public CborObject toCbor() {
+        Map<String, Cborable> map = new HashMap<>();
+        map.put("indexEntries", CborObject.CborMap.build(this.entries.entrySet().stream()
```

Comment:
🟢 
Maybe we should put `indexEntries` to some private const in this file (in a similar way as you did `TAGS_INDEX_FILENAME` )?

Reply:
Done :)